ShopperInsights/src/main/java/com/braintreepayments/api/EligiblePaymentsApi.kt

```diff
@@ -1,28 +1,34 @@
 package com.braintreepayments.api
 
 import com.braintreepayments.api.EligiblePaymentsApiRequest.Companion.toJson
+import org.json.JSONException
 
-internal class EligiblePaymentsApi {
-    fun execute(request: EligiblePaymentsApiRequest): EligiblePaymentsApiResult {
-        request.toJson()
-        // TODO: Network call
-
-        // Hardcoded result
-        return EligiblePaymentsApiResult(
-            eligibleMethods = EligiblePaymentMethods(
-                paypal = EligiblePaymentMethodDetails(
-                    canBeVaulted = true,
-                    eligibleInPayPalNetwork = true,
-                    recommended = true,
-                    recommendedPriority = 1
-                ),
-                venmo = EligiblePaymentMethodDetails(
-                    canBeVaulted = true,
-                    eligibleInPayPalNetwork = true,
-                    recommended = true,
-                    recommendedPriority = 1
-                )
-            )
+internal class EligiblePaymentsApi(
+    private val braintreeClient: BraintreeClient
+) {
+    fun execute(request: EligiblePaymentsApiRequest, callback: EligiblePaymentsCallback) {
+        val jsonBody = request.toJson()
+        // TODO: Move url to PaypalHttpClient class when it is created
+        val url = "https://api.sandbox.paypal.com/v2/payments/find-eligible-methods"
+        braintreeClient.sendPOST(
+            url,
+            jsonBody,
+            object : HttpResponseCallback {
+                override fun onResult(responseBody: String?, httpError: Exception?) {
+                    if (responseBody != null) {
+                        try {
+                            callback.onResult(
+                                EligiblePaymentsApiResult.fromJson(responseBody),
+                                null
+                            )
+                        } catch (e: JSONException) {
+                            callback.onResult(null, e)
+                        }
+                    } else {
+                        callback.onResult(null, Exception("null body"))
```

Comment:
Do we return a generic `Exception` in other API calls when the reponseBody is null? It might be good to create a specific exception type for this.

Reply:
Good point, let me see what we do elsewhere, maybe it already exists.

Reply:
Changed to match other api classes. 73740f7a38912977469a9f0e964ebf2072a4db13

They all return httpError if responseBody is null.